// Copyright 2019 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// =============================================================================

syntax = "proto3";

package tensorflow.data_validation;

// TODO(martinz): consider adding an environment.
// TODO(martinz): consider adding a LifecycleStage.
message ReasonFeatureNeeded {
  // If there is an issue in creating the field, the comment should help
  // explain why.
  // Example: "This is needed for transform XYZ (see /A/B/C)"
  string comment = 1;
}

message ReasonFeatureNeededList {
  repeated ReasonFeatureNeeded reason_feature_needed = 1;
}

// A proto representation of tensorflow::data_validation::FeaturesNeeded
// used to for serialization and deserialization.
message FeaturesNeededProto {
  // Key string are expected to be instances of
  // metadata::v0::Path::SerializeAsString()
  map<string, ReasonFeatureNeededList> feature_needed = 1;
}
